apps/server/src/modules/tool/tool-launch/controller/tool-launch.controller.ts

```diff
@@ -24,17 +29,47 @@ export class ToolLaunchController {
 	@ApiOkResponse({ description: 'Tool launch request', type: ToolLaunchRequestResponse })
 	@ApiUnauthorizedResponse({ description: 'Unauthorized' })
 	@ApiForbiddenResponse({ description: 'Forbidden' })
-	@ApiBadRequestResponse({ description: 'Outdated tools cannot be launched' })
-	async getToolLaunchRequest(
+	@ApiUnprocessableEntityResponse({ description: 'Outdated tools cannot be launched' })
+	async getContextExternalToolLaunchRequest(
 		@CurrentUser() currentUser: ICurrentUser,
-		@Param() params: ToolLaunchParams
+		@Param() params: ContextExternalToolLaunchParams
 	): Promise<ToolLaunchRequestResponse> {
-		const toolLaunchRequest: ToolLaunchRequest = await this.toolLaunchUc.getToolLaunchRequest(
+		const toolLaunchRequest: ToolLaunchRequest = await this.toolLaunchUc.getContextExternalToolLaunchRequest(
 			currentUser.userId,
 			params.contextExternalToolId
 		);
 
 		const response: ToolLaunchRequestResponse = ToolLaunchMapper.mapToToolLaunchRequestResponse(toolLaunchRequest);
 		return response;
 	}
+
+	@Post('school/:schoolExternalToolId/launch')
+	@HttpCode(HttpStatus.OK)
+	@ApiOperation({ summary: 'Get tool launch request for a context external tool id' })
```

Comment:
saw this in vue and just wanted to raise awareness ;) Not sure if swapping 'context' with 'school' would be enough , but you'll figure it out :P